File: src/test/java/net/slipp/service/user/PasswordGeneratorTest.java
package net.slipp.service.user;

import static org.hamcrest.CoreMatchers.*;
import static org.junit.Assert.*;

import org.junit.Test;

public class PasswordGeneratorTest {
    @Test
    public void randomGenerate() throws Exception {
        PasswordGenerator dut = new RandomPasswordGenerator();
        String first = dut.generate();
        String second = dut.generate();
        assertThat(first.equals(second), is(false));
    }
    
    @Test
    public void fixedGenerate_passPassword() throws Exception {
        String password = "password";
        PasswordGenerator dut = new FixedPasswordGenerator(password);
        assertThat(dut.generate(), is(password));
    }
    
    @Test
    public void fixedGenerate_null() throws Exception {
        String password = "password";
        PasswordGenerator dut = new FixedPasswordGenerator();
        assertThat(dut.generate(), is(password));
    }
}

File: src/test/scala/net/slipp/service/user/PasswordGeneratorTest.scala
package net.slipp.service.user

import org.hamcrest.CoreMatchers._
import org.junit.Assert._
import org.junit.Test

class PasswordGeneratorTest {
  @Test
  @throws(classOf[Exception])
  def randomGenerate {
    val dut: PasswordGenerator = new RandomPasswordGenerator
    val first: String = dut.generate
    val second: String = dut.generate
    assertThat(first == second, is(false))
  }

  @Test
  @throws(classOf[Exception])
  def fixedGenerate_passPassword {
    val password: String = "password"
    val dut: PasswordGenerator = new FixedPasswordGenerator(password)
    assertThat(dut.generate, is(password))
  }

  @Test
  @throws(classOf[Exception])
  def fixedGenerate_null {
    val password: String = "password"
    val dut: PasswordGenerator = new FixedPasswordGenerator
    assertThat(dut.generate, is(password))
  }
}